handlers/soft-opt-in-consent-setter/src/main/scala/com/gu/soft_opt_in_consent_setter/HandlerIAP.scala

```diff
@@ -242,7 +242,16 @@ object HandlerIAP extends LazyLogging with RequestHandler[SQSEvent, Unit] {
             logger.info(
               s"(cancellation) Sending consents request for identityId $identityId with payload: $consentsBody",
             )
-            sendConsentsReq(identityId, consentsBody)
+            sendConsentsReq(identityId, consentsBody) match {
+              case Left(notFoundErr) if notFoundErr.statusCode.contains(404) =>
+                logger.warn(s"(cancellation) Consents request for $identityId failed with 404 Not Found")
+                Right(())
+              case Left(otherError) =>
+                // Propagate all other errors
+                Left(otherError)
+              case Right(_) =>
+                Right(())
+            }
```

Comment:
This whole function is a bit lumpy even before your change, perhaps this would be more concise end result?
```suggestion
    def sendCancellationConsents(identityId: String, consents: Set[String]): Either[SoftOptInError, Unit] = {
      val maybeError: Option[SoftOptInError] =
        for {
          con <- Some(consents).filter(_.nonEmpty)
          consentsBody = consentsCalculator.buildConsentsBody(con.map(_ -> false).toMap)
          _ = logger.info(
            s"(cancellation) Sending consents request for identityId $identityId with payload: $consentsBody",
          )
          err <- sendConsentsReq(identityId, consentsBody).swap.toOption
          if !err.statusCode.contains(404)
        } yield err
      maybeError.toLeft(())
    }
```

Reply:
Thank you @johnduffell for taking the time to refactor this function 🙏